src/course_bot/essay.clj

```diff
@@ -103,13 +111,51 @@
     (talk/def-talk db cmd help
       :start
       (fn [tx {{id :id} :from}]
-        (let [submitted? (codax/get-at tx [id :essays essay-code :text])]
-          (when submitted?
-            (talk/send-text token id (format (tr :essay/your-essay-already-uploaded-:essay-name) essay-code))
-            (talk/stop-talk tx))
-          (talk/send-text token id (str (format (tr :essay/send-essay-text-in-one-message-:essay-name) essay-code)
-                                        (when topics-msg (str (tr :essay/themes) topics-msg))))
-          (talk/change-branch tx :submit)))
+        (let [submitted? (codax/get-at tx [id :essays essay-code :text])
+              reviews (codax/get-at tx [id :essays essay-code :received-review])]
+          (cond
+            ;; If the essay has reviews, warn about reuploading
+            (and submitted? (not-empty reviews))
```

Comment:
Better to prohibit reupload if we already assign some reviewers. E.g. reviewer can be in progress.